src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -161,12 +173,16 @@ private static ParquetBufferValue parseColumnValueToParquet(
             subColumnFinder,
             defaultTimezone,
             insertRowsCurrIndex,
-            isDescendantsOfRepeatingGroup);
+            isDescendantsOfRepeatingGroup,
+            error);
       }
     }
 
     if (value == null) {
       if (type.isRepetition(Repetition.REQUIRED)) {
+        if (error != null) {
+          error.addNullValueForNotNullColName(subColumnFinder.getDotPath(id));
```

Comment:
There is a critical gap that we're introducing here. Lets say a map evolves have three columns as nullable. It will take three rounds of insertRow -> get insertError with one field path called out -> alter table -> insert Row again -> get insertError with a second field path called out -> and so on.

Similarly for the other kinds of errors (extra columns / missing columns).

The right way to fix this is to only throw at the end after insertRow has processed all columns and collected all errors in the error object you're passing around. Instead of throwing on the first problem encountered.

Reply:
Deferred all exception after the whole row is parsed. Added a IT of this.